source/kraken/fill_disruption_from_chaos.cpp

```diff
@@ -46,6 +48,48 @@ namespace nt = navitia::type;
 namespace bt = boost::posix_time;
 namespace bg = boost::gregorian;
 
+namespace {
+template<typename Container, typename Pred>
+void erase_remove_if_helper(Container& c, Pred p) {
+    c.erase(std::remove_if(std::begin(c), std::end(c), p), std::end(c));
+}
+template<typename Container, typename Pred>
+void erase_remove_helper(Container& c, Pred p) {
+    c.erase(std::remove(std::begin(c), std::end(c), p), std::end(c));
+}
+
+struct DiscreteVJTag{};
```

Comment:
I have two solutions for that, which one do you guys prefer?

```c++
template<typename T>
std::vector<std::unique_ptr<T>>& get_vj_list_of_jp_helper(nt::JourneyPattern* jp);

template<>
auto get_vj_list_of_jp_helper<nt::DiscreteVehicleJourney>(nt::JourneyPattern* jp)
-> decltype((jp->discrete_vehicle_journey_list))
{ return jp->discrete_vehicle_journey_list; }

template<>
auto get_vj_list_of_jp_helper<nt::FrequencyVehicleJourney>(nt::JourneyPattern* jp)
-> decltype((jp->frequency_vehicle_journey_list))
{return jp->frequency_vehicle_journey_list;}
```

and second one:

```c++ 
template<typename T,
         typename = typename std::enable_if<std::is_same<T, nt::DiscreteVehicleJourney>::value>::type
        >
auto get_vj_list_of_jp_helper(nt::JourneyPattern* jp)
-> decltype((jp->discrete_vehicle_journey_list)) {
    return jp->discrete_vehicle_journey_list;
}

template<typename T,
         typename = typename std::enable_if<std::is_same<T, nt::FrequencyVehicleJourney>::value>::type
        >
auto get_vj_list_of_jp_helper(nt::JourneyPattern* jp)
-> decltype((jp->frequency_vehicle_journey_list)) {
    return jp->frequency_vehicle_journey_list;
}
```